pvlib/iotools/solcast.py

```diff
@@ -0,0 +1,443 @@
+        where applicable. See variable :const:`VARIABLE_MAP`.
+    kwargs:
+        Optional parameters passed to the API.
+        See [2]_ for full list of parameters.
+
+    Returns
+    -------
+    data : pandas.DataFrame
+        containing the values for the parameters requested.The times
+        in the DataFrame index indicate the midpoint of each interval.
+
+    Examples
+    --------
+    >>> pvlib.iotools.solcast.get_solcast_tmy(
+    >>>      latitude=-33.856784,
+    >>>      longitude=151.215297,
+    >>>      api_key="your-key"
+    >>> )
+
+    you can pass any of the parameters listed in the API docs,
+    like ``time_zone``. Here we set the value of 10 for
+    "10 hours ahead of UTC":
+
+    >>> pvlib.iotools.solcast.get_solcast_tmy(
+    >>>     latitude=-33.856784,
+    >>>     longitude=151.215297,
+    >>>     time_zone=10,
+    >>>     api_key="your-key"
+    >>> )
+
+    References
+    ----------
+    .. [1] `Get an API Key <https://toolkit.solcast.com.au/register>`_
+    .. [2] `Solcast API Docs <https://docs.solcast.com.au/>`_
+    """
+
+    params = dict(
+        latitude=latitude,
+        longitude=longitude,
+        format="json",
+        **kwargs
+    )
+
+    data = _get_solcast(
+        endpoint="tmy/radiation_and_weather",
+        params=params,
+        api_key=api_key,
+        map_variables=map_variables
+    )
+
+    return data, {}
+
+
+def get_solcast_historic(
+    latitude,
+    longitude,
+    start,
+    api_key,
+    end=None,
+    duration=None,
+    map_variables=True,
+    **kwargs
+):
+    """Get historical irradiance and weather estimated actuals
+
+    for up to 31 days of data at a time for a requested location,
+    derived from satellite (clouds and irradiance
+    over non-polar continental areas) and
+    numerical weather models (other data).
+    Data is available from 2007-01-01T00:00Z up to real time estimated actuals.
+
+    Parameters
+    ----------
+    latitude : float
+        in decimal degrees, between -90 and 90, north is positive
+    longitude : float
+        in decimal degrees, between -180 and 180, east is positive
+    start : datetime-like
+        First day of the requested period
+    end : optional, datetime-like
+        Last day of the requested period.
+        Must include one of ``end`` or ``duration``.
+
+    duration : optional, default is None
+        Must include either  ``end`` or ``duration``.
+        ISO_8601 compliant duration for the historic data,
+        like "P1D" for one day of data.
+        Must be within 31 days of the start_date.
+    map_variables: bool, default: True
+        When true, renames columns of the DataFrame to pvlib variable names
+        where applicable. See variable :const:`VARIABLE_MAP`.
+    api_key : str
+        To access Solcast data you will need an API key [1]_.
+    kwargs:
+        Optional parameters passed to the GET request
+
+    See [2]_ for full list of parameters.

```

Comment:
```suggestion
    kwargs:
        Optional parameters passed to the API.
        See [2]_ for full list of parameters.
```

For `kwargs`, we should be sure to be clear about the difference between GET parameters and parameters to `requests.get`.   Same for the other functions

Reply:
done.